app/src/main/java/com/example/composetemplate/data/remote/analytics/AnalyticsModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.composetemplate.data.remote.analytics
+
+import com.example.composetemplate.data.remote.analytics.AnalyticsManager.logAnalytics
+
+class AnalyticsParameters(
+    val event: Events,
+    val itemName: String? = null,
+    val value: String? = null,
+) {
+
+    enum class Events {
+        SELECT_LECTURE;
+
+        val rawValue: String
+            get() = when (this) {
+                SELECT_LECTURE -> this.name.lowercase()
+            }
+    }
+}
+
+class AnalyticsLogger {
```

Comment:
Why do we need this class? We already have the singleton `AnalyticsManager`. Why can't we just use it?  
You could add a new public function to `AnalyticsManager` that performs a `when` operation over all the `Event` enum values. For example:

fun <T> logEvent(data: T, event: Event) {
    when (event) {
        SELECT_LECTURE -> logAnalytics(AnalyticsParameters(data as? String, event))
        SELECT_SOMETHING -> logAnalytics(AnalyticsParameters(data as? Int, event))
    }
}

I think that's better way instead of creating different function for each event.

By the way, I think `AnalyticsParameters` should be a data class.

Reply:
good notice!